src/mds/Locker.cc

```diff
@@ -2706,7 +2708,7 @@ void Locker::handle_client_caps(MClientCaps *m)
     }
 
     // filter wanted based on what we could ever give out (given auth/replica status)
-    bool need_flush = false;
+    bool need_flush = m->flags & CLIENT_CAPS_SYNC;
```

Comment:
To make it clearer that we're always respecting this flag, maybe the need_flush assignments below should be changed into `need_flush |=`?